src/elements/src/index.ts

```diff
@@ -0,0 +1,55 @@
+import { LifeCycles } from 'single-spa';
+import { NgElement } from '@angular/elements';
+
+import {
+  SingleSpaAngularElementsOptions,
+  BootstrappedSingleSpaAngularElementsOptions,
+} from './types';
+
+const defaultOptions: BootstrappedSingleSpaAngularElementsOptions = {
+  element: null!,
+  ngModuleRef: null,
+  bootstrapFunction: null!,
+  withProperties: undefined,
+  domElementGetter: undefined,
+};
+
+async function bootstrap(options: BootstrappedSingleSpaAngularElementsOptions) {
+  if (options.ngModuleRef !== null) {
+    return;
+  }
+
+  options.ngModuleRef = await options.bootstrapFunction();
+}
+
+async function mount(options: BootstrappedSingleSpaAngularElementsOptions) {
+  if (typeof options.withProperties === 'function') {
+    const properties = await options.withProperties(options.ngModuleRef!);
```

Comment:
I think it would make sense to pass the single-spa props as the properties on the custom element.